low-level/framework/utils/ipmi_client.py

```diff
@@ -161,39 +185,67 @@ def get_fru_list_by_type(self, fru_list, sensor_id_map):
                 for fru in fru_detail}
         return sensor_id_map
 
-    def _run_command(self, command, out_file=subprocess.PIPE):
-        """executes commands"""
-        process = subprocess.Popen(command, shell=True, stdout=out_file, stderr=subprocess.PIPE)
-        result = process.communicate()
-        return result, process.returncode
+    def _run_ipmitool_subcommand(self, subcommand, grep_args=None):
+        """Executes ipmitool sub-commands, and optionally greps the output."""
+        self.ACTIVE_IPMI_TOOL = self.IPMITOOL
+        host_conf_cmd = ""
 
-    def _run_ipmitool_subcommand(self, subcommand, grep_args=None, out_file=subprocess.PIPE):
-        """executes ipmitool sub-commands, and optionally greps the output"""
+        # Set ipmitool to ipmisimtool if activated.
+        if os.path.exists("/tmp/activate_ipmisimtool"):
+            cmd = self.IPMISIMTOOL + " sel info"
+            _, _, retcode = SimpleProcess(cmd).run()
+            if retcode in [0, 2]:
+                self.ACTIVE_IPMI_TOOL = self.IPMISIMTOOL
+                logger.info("IPMI simulator is activated.")
 
-        ipmi_tool = self.IPMITOOL
+        # Fetch channel info from config file and cache.
+        _channel_interface = Conf.get(SSPL_CONF, "%s>%s" %
+                                (BMC_INTERFACE, BMC_CHANNEL_IF))
 
-        # A dummy file path check to select ipmi simulator if
-        # simulator is required, otherwise default ipmitool.
-        if os.path.exists("/tmp/activate_ipmisimtool"):
-            res, retcode = self._run_command(command=f"{self.IPMISIMTOOL} sel info")
-            if retcode == 0:
-                ipmi_tool = self.IPMISIMTOOL
-                logger.info("IPMI simulator is activated")
-
-        command = ipmi_tool + subcommand
-        if grep_args is not None:
-            command += " | grep "
-            if isinstance(grep_args, list):
-                grep_args_str = ""
-                for arg in grep_args:
-                    grep_args_str = "'{}' ".format(arg)
-                command += grep_args_str
-            else:
-                command += "'{}'".format(grep_args)
-        res, retcode = self._run_command(command, out_file)
+        _active_interface = store.get(self.ACTIVE_INTERFACE, None)
+
+        # Set host_conf_cmd based on channel info.
+        if _channel_interface == self.LAN_IF and \
```

Comment:
1. `Can be done in initialize() and use it everytime then` 
sure but reading from local class variable and from conf_store cache is exactly same (except in conf_store we are making a  get function call instead of directly accessing the class variable ), I thought this would be better for future, should I initialize it in \_new\_()  method, that's when the first obj of singleton will be spawned.

2. `active interface should match LAN_IF`
sure let me change this.

Reply:
>sure but reading from local class variable and from conf_store cache is exactly same (except in conf_store we are making a get function call instead of directly accessing the class variable ), I thought this would be better for future,

I liked the readiness from future perspective and reading from confStore in each iteration would serve that good. So when dynamic cfg support is implemented, we dont need to change code here